docs/src/IconButton.doc.js

```diff
@@ -1,419 +1,862 @@
       },
       {
         name: 'selected',
         type: 'boolean',
-        required: false,
-        defaultValue: false,
-        description: [
-          'Control the "selected" state of a button component to toggle binary states.',
-          'Optional with button-role + button-type buttons.',
-        ],
-        href: 'selected',
+        description:
+          'Toggles between binary states: on/off, selected/unselected, open/closed. See the [selected](#Selected-state) variant to learn more.',
       },
+    ]}
+  />,
+);
+
+card(
+  <PropTable
+    Component={IconButton}
+    name="Additional role = link"
+    id="role_link"
+    props={[
       {
-        name: 'size',
-        type: `"xs" | "sm" | "md" | "lg" | "xl"`,
-        required: false,
-        defaultValue: 'md',
+        name: 'role',
+        type: 'link',
+        required: true,
+        description:
+          'Sets link interaction in the component. See the [role](#Role) variant and [OnLinkNavigationProvider](/OnLinkNavigationProvider) to learn more about link navigation.',
+      },
+      {
+        name: 'href',
+        type: 'string',
+        required: true,
+        description: 'Specifies a link URL.',
+      },
+      {
+        name: 'rel',
+        type: `'none' | 'nofollow'`,
         description:
-          'Display an IconButton in different sizes. Size changes the component padding modifying its fixed height & width: xs (24px), sm (32px), md (40px), lg (48px), xl (56px).',
-        href: 'size',
+          'Specifies the relationship between the current document and the linked document. See the [role](#Role) variant to learn more.',
       },
       {
         name: 'target',
         type: `null | 'self' | 'blank'`,
-        required: false,
-        defaultValue: 'null',
-        description: [
-          'Define the frame or window to open the anchor defined on `href`:',
-          `- 'null' opens the anchor in the same window.`,
-          `- 'blank' opens the anchor in a new window.`,
-          `- 'self' opens an anchor in the same frame.`,
-          'Optional with link-role buttons.',
-        ],
-        href: 'roles',
+        description: `Define the frame or window to open the anchor defined on \`href\`. See the [role](#Role) variant to learn more.`,
       },
     ]}
   />,
 );
 
 card(
-  <Example
-    name="Basic IconButton"
-    id="basic"
-    defaultCode={`
-<IconButton
-  accessibilityLabel="Open edit modal to edit Board"
-  icon="edit"
-/>
-`}
-  />,
+  <MainSection name="Usage guidelines">
+    <MainSection.Subsection columns={2}>
+      <MainSection.Card
+        cardSize="md"
+        type="do"
+        title="When to Use"
+        description={`
+- Interface space is limited. Prefer using a [Button](/Button) if space is available.
+- Triggering a [Modal](/Modal) to complete a related task.
+- Creating visual separation of actions in text-heavy contents.
+- Less-emphasis actions that don't impede users from completing a task.
```

Comment:
Small language tweaks:

- Interface space is limited. Prioritize using a [Button](/Button) if space is available.
- Triggering a [Modal](/Modal) to complete a related task.
- Creating visual separation of actions in text-heavy content.
- Lower-emphasis actions that don't impede users from completing a task.

Reply:
It looks better! Thanks, Ayesha ;)

Reply:
DONE